### ISSUE TITLE
Feature Request: Update via commandline

### ISSUE THREAD

**Issue by DocMAX** [doZennn/steam-rom-manager#431]
I frequently start SRM just to do one thing: Update the ROM list and exit.
Since this is the same task every time i would automate this with a cron job / systemd timer or something.
So the only thing i would need is a oneliner without GUI like "steam-rom-manager -update -quiet" or something.
Can this be done maybe?

Thanks, DocMAX

**Comment by marissa999** [comment]
I love that idea, it would also make it much easier for Steam Deck User. Instead of having to go into Desktop mode, closing Steam, starting SRM, and doing the parse+update process you could just do it through SSH. Maybe somebody could then even write a Docky-Plugin to trigger that process through the Gamemode UI directly, meaning that all that is left to do would be restarting the Deck.

**Comment by DocMAX** [comment]
Glad i'm not the only one. My ULTIMATE goal of this thing is to just "drop" games into a "watched" gamefolder and the games imedatelly show up in Steam gamelist (an vice versa). But i think we are far away from that, right?

Game put in game folder -> SRM gets notified (inotify?), starts it scan (with my personal script to find machine type, emulator, exeutable and so on) -> Refreshes Steam game list *while running (in GamepadUI mode)* -> Done

**Comment by natopotato390** [comment]
👍want this

**Comment by Godsbane** [comment]
Duplicate of #58 :)

**Comment by Sanfe** [comment]
It will be great to create a systemd service so we could update in boot without need to enter in desktop mode. It's feasible to copy roms with syncthing.

**Comment by cbartondock** [comment]
I agree, but this is a dupe, closing

**Comment by DocMAX** [comment]
Our dream came true!

**Comment by cbartondock** [comment]
You're welcome 🥳

**Comment by DocMAX** [comment]
But just a little remark:
![grafik](https://user-images.githubusercontent.com/5351323/233506392-64bcc7f5-ef6f-4392-a867-742c90e089d0.png)
Why is it depending on X11? I want to scan on command line only.

**Comment by cbartondock** [comment]
`export DISPLAY=:0; npx electron . list 2> /dev/null` worked fine for me running it over SSH. 

The reason is that the command line interface is really just calling functions in the renderer process via an ipc channel. It spins up the whole GUI process in the background. Not very efficient I know, but much easier to maintain.

**Comment by cbartondock** [comment]
Replace `npx electron .` with your SRM executable
